### DISCUSSION TITLE
Adding playlist to Plex/Spotify

### DISCUSSION THREAD

**Discussion by arsaboo** [beetbox/beets#4410] [Ideas]
After the recent PR (#4399) by @JOJ0, it would be great to be able to create these playlists in Plex or Spotify. Leveraging beets intelligence in creating smart playlists on these popular platforms will greatly improve music consumption. 

For Plex, we have [plexapi](https://github.com/pkkid/python-plexapi) that provides all the required endpoints to create/modify in Plex. Similarly, [Spotipy](https://github.com/plamere/spotipy) provides all the endpoints for interacting with Spotify.

**Comment by arsaboo** [comment]
I just came across [Calliope](https://gitlab.com/samthursfield/calliope/) that may be of relevance here.

**Comment by arsaboo** [comment]
@sampsyo I am working on extending the Plex plugin to add additional features. This would be based on the `plexapi` package. I was wondering if we should continue to use the current `plexupdate` plugin name and add additional subcommands there. This would prevent any breaking changes. The downside would be that the plugin name may not capture all the features and that I am not well versed with tests (so will have to remove those).  

I have a [version ready](https://github.com/arsaboo/beets/blob/plex_update/beetsplug/plexsync.py) with `plexsync` name ready to go. If it works for you, I can create an initial PR to move `plexupdate` plugin to use `plexapi` without losing any current functionality and adds the functionality to get Plex (userRating) information in beets.

**Reply by sampsyo** [reply]
Good question! Maybe the simplest strategy would be to make a second plugin with the new functionality? They can both share the same configuration—in particular, since the existing plugin uses a namespace called `plex:` instead of `plexupdate:` or similar:
https://github.com/beetbox/beets/blob/50825dbfc8443869ff2f1cad406e310a597c1aad/beetsplug/plexupdate.py#L78

…it should be easy for users to just enable both plugins.

**Reply by arsaboo** [reply]
Also, I have included the update feature (which can also be manually invoked with `beet plexupdate` command) in the new plugin. I guess, we can have both the plugins for some time and then deprecate the `plexupdate` plugin.

**Comment by arsaboo** [comment]
Ok...created a PR https://github.com/beetbox/beets/pull/4418
